applications/FluidDynamicsApplication/python_scripts/apply_wall_law_process.py

```diff
@@ -0,0 +1,152 @@
+        helper_class_name = f"{wall_model_name}_helper"
+        if not hasattr(self, helper_class_name):
+            err_msg = f"Wrong wall model helper class name '{helper_class_name}'. Expected in this case is '__{wall_model_name}_helper'. Check your implementation."
+            raise Exception(err_msg)
+        self.wall_model_helper = getattr(self, helper_class_name)()
+
+        # Validate the wall model settings with the corresponding wall model defaults
+        Settings["wall_model_settings"].ValidateAndAssignDefaults(self.wall_model_helper.GetWallModelDefaultSettings())
+
+        # Save auxiliary member variables
+        self.model = Model
+        self.settings = Settings
+
+    def GetDefaultParameters(self):
+        default_settings = KratosMultiphysics.Parameters("""{
+            "model_part_name" : "",
+            "wall_model_name" : "",
+            "wall_model_settings" : {},
+            "calculate_normals_at_each_step" : false
+        }""")
+        return default_settings
+
+    def ExecuteInitialize(self):
+        # Get data from settings
+        model_part =  self.model[self.settings["model_part_name"].GetString()]
+        wall_model_name = self.settings["wall_model_name"].GetString()
+
+        # Get the condition registry name
+        # Note that in here we are assuming a unique condition geometry in model part
+        pts_num = 0
+        for condition in model_part.Conditions:
+            pts_num = condition.GetGeometry().PointsNumber()
+            break
+        pts_num = model_part.GetCommunicator().GetDataCommunicator().MaxAll(pts_num)
+        domain_size = model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        cond_aux_name = self.__wall_model_condition_name_map[wall_model_name]
+        cond_reg_name = f"{cond_aux_name}{domain_size}D{pts_num}N"
+
+        # Substitute current conditions by the corresponding ones implementing the wall model
+        # 1. Save current condition data in a map such that Id : (geometry, properties)
+        aux_cond_data_map = {}
+        for condition in model_part.Conditions:
+            condition.Set(KratosMultiphysics.TO_ERASE,True)
+            aux_ids = [node.Id for node in condition.GetNodes()]
+            aux_props = condition.Properties
+            aux_cond_data_map[condition.Id] = (aux_ids, aux_props)
+        # 2. Remove current conditions
+        # Note that in here we need to assume that no submodelpart is hanging from this one as the hierarchical structure would be destroyed
+        if model_part.NumberOfSubModelParts() != 0:
+            raise Exception(f"Wall model part '{model_part.FullName}' has submodelparts.")
+        model_part.RemoveConditionsFromAllLevels(KratosMultiphysics.TO_ERASE)
+        # 3. Create new wall law conditions from the data map info
+        for key, value in aux_cond_data_map.items():
+            model_part.CreateNewCondition(cond_reg_name, key, value[0], value[1])
```

Comment:
Is there a specific reason not using `ReplaceElementsAndConditionsProcess`?

Reply:
Nope. Done!